### PR TITLE
refactor: move to create nodes v2

### LINKED ISSUE #7: use createNodesV2 for plugins
Use [createNodesV2](https://nx.dev/nx-api/devkit/documents/CreateNodesV2) instead of [createNodes](https://nx.dev/nx-api/devkit/documents/CreateNodes) in all plugins

### PR DESCRIPTION
**Pull request by BioPhoton**
close #7

### CODE REVIEW

File: projects/build-env/src/plugin/targets/environment.targets.ts [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+export function updateEnvTargetNames(
+  projectConfig: ProjectConfiguration,
+  options: Required<Pick<BuildEnvEnvironmentsOptions, 'targetNames'>>
+): Record<string, TargetConfiguration> {
+  const { targetNames: envTargetNames } = options;
+  const { targets: existingTargets = {} as TargetConfiguration } =
+    projectConfig;
+
+  return Object.fromEntries(
+    Object.entries(existingTargets).map(([existingTargetName, config]) => {
```
Comment (getlarge) [inline projects/build-env/src/plugin/targets/environment.targets.ts:126]: maybe that's acceptable to use intermediary variables to make it more readable?

File: projects/build-env/src/plugin/targets/environment.targets.ts [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+  options: Pick<
+    NormalizedCreateNodeOptions['environments'],
+    'environmentsDir'
+  > &
+    Omit<StartVerdaccioOptions, 'projectName'>
+): Record<string, TargetConfiguration> {
+  const { name: envProject } = projectConfig;
+  const { environmentsDir, ...verdaccioOptions } = options;
+  const environmentDir = join(environmentsDir, envProject);
+
```
Comment (getlarge) [inline projects/build-env/src/plugin/targets/environment.targets.ts:58]: would this make sense to check in this function if project is of type 'e2e'?

File: projects/build-env/src/plugin/normalize-create-nodes-options.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  packages: BuildEnvPackagesOptions;
+};
+
+export function normalizeCreateNodesOptions(
+  options: BuildEnvPluginCreateNodeOptions
+): NormalizedCreateNodeOptions {
+  const {
+    environments: givenEnvironments = {},
+    packages: givenPublishable = {},
+  } = options ?? {};
```
[the remark this answers is not in the document]
Comment (BioPhoton) [reply projects/build-env/src/plugin/normalize-create-nodes-options.ts:28]: Good catch ´! It is a left over and i'll remove them.

File: projects/build-env/src/plugin/targets/environment.targets.ts [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+  return {
+    [DEFAULT_START_VERDACCIO_TARGET]: {
+      // @TODO: consider using the executor function directly to reduce the number of targets
+      executor: '@nx/js:verdaccio',
```
[the remark this answers is not in the document]
Comment (BioPhoton) [reply projects/build-env/src/plugin/targets/environment.targets.ts:62]: This: https://github.com/nrwl/nx/blob/b73f1e0e0002c55fc0bacaa1557140adb9eec8de/packages/js/src/executors/verdaccio/verdaccio.impl.ts#L22

Will add it to the comment

File: projects/build-env/src/plugin/targets/environment.targets.ts (hunk 1 above)
Comment (BioPhoton) [reply projects/build-env/src/plugin/targets/environment.targets.ts:126]: The function `isEnvProject` is used to check if `updateEnvTargetNames` should execute. Is this what you ask for?

File: projects/build-env/src/plugin/targets/environment.targets.ts (hunk 2 above)
Comment (BioPhoton) [reply projects/build-env/src/plugin/targets/environment.targets.ts:58]: I guess it should not be a strong requirement.

File: projects/build-env/src/plugin/normalize-create-nodes-options.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (BioPhoton) [reply projects/build-env/src/plugin/normalize-create-nodes-options.ts]: removed

File: projects/build-env/src/plugin/targets/environment.targets.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (BioPhoton) [reply projects/build-env/src/plugin/targets/environment.targets.ts]: added

File: projects/build-env/src/plugin/targets/environment.targets.ts [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+        port: uniquePort(),
+        storage: join(environmentDir, 'storage'),
+        clear: true,
+        environmentDir,
+        projectName: envProject,
+        ...verdaccioOptions,
+      },
+    },
+    [DEFAULT_STOP_VERDACCIO_TARGET]: {
+      executor: '@push-based/build-env:kill-process',
```
[the remark this answers is not in the document]
Comment (BioPhoton) [reply projects/build-env/src/plugin/targets/environment.targets.ts]: use constants now

File: projects/build-env/src/plugin/targets/environment.targets.ts (hunk 1 above)
Comment (BioPhoton) [reply projects/build-env/src/plugin/targets/environment.targets.ts]: Ah now I get it!! I can try

Comment (BioPhoton) [reply projects/build-env/src/plugin/targets/environment.targets.ts]: So..

I created a helper:

```ts
export function iterateEntries<T, K extends string, V>(
  obj: T,
  cb: (entries: [keyof T, T[keyof T]][]) => [K, V][]
): { [key in K]: V } {
  return Object.fromEntries(
    cb(Object.entries(obj) as [keyof T, T[keyof T]][])
  ) as { [key in K]: V };
}
```

and use it like this:
```ts

  return iterateEntries(existingTargets, (entries) => entries
    .map(([existingTargetName, config]) => {
        // ...
        return [existingTargetName, config];
    })
  );
```

**Pull request merged**